neuralmonkey/model/model_part.py

```diff
@@ -44,6 +44,34 @@ def use_scope(self):
             with tf.name_scope(self._variable_scope.original_name_scope):
                 yield
 
+    def get_dependencies(self) -> Set["ModelPart"]:
```

Comment:
To jsem právě myslel, že ta hrůza s `hasattr` zmizí, tahle metoda bude absraktní a každý `ModelPart` to udělá, jak bude sám nejlépe umí.

Reply:
To by se nám hrozně moc kopíroval kód. ModelPartů máme asi dvacet a drtivá většina z nich má encoder nebo encoders. Otázka můžou být ty attentiony, ale u enkodérů by to zasvinilo deset až dvacet souborů naprosto identickýma patnácti řádkama kódu.

Reply:
Když bude nějaká model part potřebovat větší chytristiku, může si tuhle funkci samozřejmě předefinovat.